### PR TITLE
app.nit: propagate events (such as a button press) to the parents of the control

### PR DESCRIPTION
**Pull request by xymus**
UI events, such as a button press or a check box toggle, were only sent to the observers of a control, and the control itself. For this reason, every button was watched by the window or some other container.

Propagating the events to the parents (direct and indirect) reaches the window and all of its containers. This allows for more modular code and components, and it removes the need for manually declaring the window as an observer.

### CODE REVIEW

File: contrib/benitlux/src/client/features/debug.nit [hunk 1]
```diff
@@ -24,7 +24,7 @@ redef class UserWindow
 	private var layout_debug = new VerticalLayout(parent=layout)
 
 	private var lbl_debug_title = new Label(parent=layout_debug,
-		text="Debug options".t)
+		text="Debug options".t, size=1.5)
```
Comment (jcbrinfo) [inline contrib/benitlux/src/client/features/debug.nit]: Seems out of the scope of the PR or the commit. (1 other line like this)

File: lib/app/ui.nit [hunk 2]
```diff
@@ -211,7 +230,7 @@ abstract class TextView
 	# depending on the customization options of each platform.
 	# For consistent results, it is recommended to use only on instances
 	# of `Label` and `size` should be either 0.0, 0.5 or 1.0.
-	fun align=(center: nullable Float) is autoinit do end
+	fun align=(align: nullable Float) is autoinit do end
```
Comment (jcbrinfo) [inline lib/app/ui.nit]: Seems out of the scope of the PR.

File: contrib/benitlux/src/client/features/debug.nit (hunk 1 above)
Comment (xymus) [inline contrib/benitlux/src/client/features/debug.nit]: No, it's OK. The init was left empty except for this line, I moved it up so it is cleaner. Moving a single line for aesthetic purposes as a direct side effect of this PR is not worth a dedicated PR.

File: lib/app/ui.nit (hunk 2 above)
Comment (xymus) [inline lib/app/ui.nit]: Once again, this one is a simple fix which is a no-brainer, it's not worth a dedicated PR.

**Comment by privat** [comment]
+1

**Pull request merged**